metabase/metabase
# Custom labels for number parameter

[Testing plan](https://www.notion.so/metabase/test-plan-Add-custom-list-input-support-and-labels-for-parameters-111e1ea897914b7587a35bfaa5c8b24f)

Closes https://github.com/metabase/metabase/issues/44054

### Description

Adds the ability to add custom labels to number parameters.

## Testing this PR
1. Create a dashboard with a visualization for the Accounts table
2. Create a number parameter and link it to Accounts -> Seats
3. Edit the parameter, set "How should people filter on this column?" to "Search box" and click "Edit"
4. In the modal select "Custom list" and enter
  ```
  10, Ten
  21, Twenty-one
  42
  ```
5. Click Done and save the dashboard
6. Select the parameter and in the dropdown input enter "Ten", the corresponding option should show up, select it. The value tag should show "Ten"
7. Click "Update filter"
9. The parameter widget should show "Ten"

The dashboard should be filtered with only rows with 10 seats showing up
Do the same when setting "How should people filter on this column?" to  "Dropdown list" and make sure "Ten" shows up.

Try the same for native queries with a number variable.
